src/Terminal/index.ts

```diff
@@ -54,13 +55,17 @@ const initTerminal = async (props: TerminalProps) => {
   const startTime = TelemetryProcessor.traceStart(Action.OpenTerminal, data);
 
   try {
-    await JupyterLabAppFactory.createTerminalApp(serverSettings);
+    await new JupyterLabAppFactory(() => closeTab(props.tabId)).createTerminalApp(serverSettings);
     TelemetryProcessor.traceSuccess(Action.OpenTerminal, data, startTime);
   } catch (error) {
     TelemetryProcessor.traceFailure(Action.OpenTerminal, data, startTime);
   }
 };
 
+const closeTab = (tabId: string): void => {
+  window.parent.postMessage({ type: MessageTypes.CloseTab, data: { tabId: tabId }, signature: "pcIframe" });
```

Comment:
Wont work for this usecase, since we want to set the postRobot listener's whitelisted window as that of the iframe, the iframe is not present when we want to initialize this.